yew/src/agent/mod.rs

```diff
@@ -24,9 +24,9 @@ pub trait Agent: Sized + 'static {
     /// Type of an input message.
```

Comment:
What if we bind `Reach` to `Discoverer<Agent = Self>`? Does that help anything?

Reply:
I'll give it a shot tomorrow, quite possible it might.

Reply:
This puts a stricter constraint on valid values of `Reach`, which is great, but does not automatically tie the knot for the typechecker.

Reply:
Got it, sounds good. Thanks for adding!